code/modules/virus2/effect/effect.dm

```diff
@@ -463,6 +463,60 @@
 	..()
 
 
+/datum/disease2/effect/spiky_skin
+	name = "Porokeratosis Acanthus"
+	stage = 2
+	max_count = 1
+	var/skip = FALSE
+
+/datum/disease2/effect/spiky_skin/activate(var/mob/living/carbon/mob,var/multiplier)
+	if(ishuman(mob))
+		var/mob/living/carbon/human/H = mob
+		if(H.species && (H.species.anatomy_flags & NO_SKIN))	//Can't have spiky skin if you don't have skin at all.
+			skip = TRUE
+			return
+	to_chat(mob, "<span class='warning'>Your skin feels a little prickly.</span>")
+
+/datum/disease2/effect/spiky_skin/deactivate(var/mob/living/carbon/mob)
+	if(!skip)
+		to_chat(mob, "<span class='notice'>Your skin feels nice and smooth again!</span>")
+	..()
+
+/datum/disease2/effect/spiky_skin/on_touch(var/mob/living/carbon/mob, var/toucher, var/touched, var/touch_type)
+	if(count && !skip)
+		var/datum/organ/external/E
+		var/mob/living/carbon/human/H
+		if(toucher == mob)	//we bumped into someone else
+			if(ishuman(touched))
+				H = touched
+		else	//someone else bumped into us
+			if(ishuman(toucher))
+				H = toucher
+		if(H)
+			while(!E || (E.status & ORGAN_ROBOT) || (E.status & ORGAN_PEG))
```

Comment:
On the one hand, PJB is objectively right
But on the other, holy fuck imagine how confusing it would be to new coders seeing `/mob/living/silicon`s AND `/mob/living/carbon/human/robot` or something